OsuParsers/Decoders/BeatmapDecoder.cs

```diff
@@ -9,18 +9,32 @@
 using System;
 using System.Collections.Generic;
 using System.Drawing;
+using System.IO;
 using System.Linq;
 using System.Numerics;
 
 namespace OsuParsers.Decoders
 {
-    internal class BeatmapDecoder
+    public class BeatmapDecoder
     {
         private Beatmap Beatmap;
         private Sections currentSection = Sections.None;
         private List<string> sbLines = new List<string>();
 
-        public Beatmap Decode(string[] lines)
+        /// <summary>
+        /// Parses .osu file.
+        /// </summary>
+        /// <param name="pathToBeatmap">Path to the .osu file.</param>
+        /// <returns>A usable beatmap.</returns>
+        public Beatmap Decode(string path)
+        {
+            if (File.Exists(path))
+                return Decode(File.ReadAllLines(path));
+            else
+                throw new FileNotFoundException();
+        }
+
+        public Beatmap Decode(IEnumerable<string> lines)
         {
             Beatmap = new Beatmap();
```

Comment:
I want to work on making all of these `Decoder` classes static, if possible. It would make it much easier to use for the end user.

Reply:
First off I think this `Beatmap` variable can be made obsolete by adding it to the arguments of the other methods that require it.

Reply:
Sure, sounds good. Let's see how that works out.